Userland/Libraries/LibWeb/Layout/TreeBuilder.h

```diff
@@ -19,7 +19,11 @@ class TreeBuilder {
     RefPtr<Layout::Node> build(DOM::Node&);
 
 private:
-    void create_layout_tree(DOM::Node&);
+    struct Context {
+        bool has_svg_root = false;
+    };
+
+    void create_layout_tree(DOM::Node&, Context);
```

Comment:
Let's pass `Context` by reference instead of copying it for every step of recursion.

Reply:
I thought passing it by value was so only children of an SVG node had the flag set on it. A reference would leak that to siblings.

Reply:
@AtkinsSJ That is why I decided to copy it.
@awesomekling I could pass it by reference, but then I have to set `has_svg_root = false` after the children are processed. Unfortunately there is a possible `return` between `has_svg_root = true` and `for_each_child`, so it could still leak if I don't handle it there as well.
What do you suggest?

Reply:
I suggest using a `TemporaryChange` RAII object to set and unset the flag, something like:

```cpp
TemporaryChange change(context.has_svg_root, true);
```